refactor: migrate to tailwind v4

dandedotdev opened the pull request:
Thank you for developing the template. I really appreciate it.

Recently, I spent some time migrating my website to Tailwind v4 so I think maybe I can try contributing to the template.

I put these styles at the `utilities` layer because the `prose` provided by `@tailwind/typography` is also at this layer so that would be easier.

Thank you for your time and attention.

timlrx commented:
Thanks for the PR. Seems like the changes might have affected the code block styles.

New:
![CleanShot 2025-02-23 at 15 55 01](https://github.com/user-attachments/assets/35b8b91b-83cc-4958-b262-46bc02f6acc7)

Previous:
![CleanShot 2025-02-23 at 15 55 43](https://github.com/user-attachments/assets/b87273df-8679-4bb8-a957-8bdd0cc48657)

dandedotdev commented:
I'd like to apologize for my carelessness. Thank you for checking the actual applied styles for me.

In [03a8ec5](https://github.com/timlrx/tailwind-nextjs-starter-blog/pull/1122/commits/03a8ec5c458d503ba722c1079835d3c228644e28), I revise the selector for `code`:

```css
& :where(code):not(.code-highlight)
```

I think it might be better than (from the perspective of specificity):

```css
& :where(code):not(pre code)
```

However, if the template switches to another code highlighting tool, this might not work.

I also took the screenshots:

| Light Theme | Dark Theme |
| --- | --- |
| ![Screenshot 2025-02-23 23 23 28](https://github.com/user-attachments/assets/6abd732b-8497-4346-99a6-2edf74785656) | ![Screenshot 2025-02-23 23 23 10](https://github.com/user-attachments/assets/357945f5-acb7-4a51-b993-6f790b871e98) |

Thank you again for your kindness and attention.

dandedotdev commented:
After scanning the output file of Tailwind v4, I found some use of `pre code`. Hence, I think it's okay.

I also corrected a mistake I made, in the content in the original `tailwind.config.js`:

```js
module.exports = {
  theme: {
    extend: {
      typography: ({ theme }) => ({
        DEFAULT: {
          // ...
        invert: { // HERE
         // ...
        },
      }),
    },
  },
  plugins: [require('@tailwindcss/forms'), require('@tailwindcss/typography')],
}
```

Is for `.prose-invert`, so I corrected it.

There are still a few differences between the Tailwind output of my PR and that of the main branch. Sadly, for now, there is no way to get a similar output of `@tailwindcss/typography` by Tailwind v4. At least, there are no style conflicts.

timlrx commented:
Thanks, this looks good to me!
